File: Java/Calculadora.java
import java.util.Map;
public class Calculadora {
    private Map<String, Operacion> operaciones;
    public Calculadora(Map<String, Operacion> operaciones){
        this.operaciones = operaciones;
    }
public String calcular(double a, double b, String simbolo) {
    Operacion operacion = operaciones.get(simbolo);
    if (operacion == null) {
        return "Operacion no valida";
    }

    try {
        double resultado = operacion.ejecutar(a, b);
        return String.valueOf(resultado);
    } catch (ArithmeticException e) {
        return e.getMessage();
    }
}

File: Python/Calculadora.py
#Dependency inversion
#solo cumple
class Calculadora:
    def __init__(self, operaciones):
        self.operaciones = operaciones
#Open/closed 
    def calcular(self, a, b, simbolo):
        operacion = self.operaciones.get(simbolo)
        if operacion is None:
            return "Operacion no válida"
        try:
        #liskov substitution
            return operacion.ejecutar(a, b)
        except ValueError as e:
            return str(e)